### DISCUSSION TITLE
FAST DDS installation error from source in Linux

### DISCUSSION THREAD

**Discussion by Jpgig** [eProsima/Fast-DDS#3365] [General]
I am trying to install FAST DDS in Linux installation from sources as per the below link:
https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html

sudo apt install cmake g++ python3-pip wget git
sudo apt install libasio-dev libtinyxml2-dev
sudo apt install libssl-dev
sudo apt install libp11-dev libengine-pkcs11-openssl
sudo apt install softhsm2
sudo usermod -a -G softhsm jiten
sudo apt install libengine-pkcs11-openssl
p11-kit list-modules
openssl engine pkcs11 -t

While running the last command, getting the below error:

![image](https://user-images.githubusercontent.com/115164852/225127891-c1d26878-788c-4edb-847a-84475f790023.png)

Any help to resolve this is appreciated.

**Comment by Jpgig** [comment]
I am able to resolve this error by putting the below lines into openssl.cnf file (located at /usr/lib/ssl)
[openssl_init]
engines = engine_section

[engine_section]
pkcs11 = pkcs11_section

[pkcs11_section]
engine_id = pkcs11
dynamic_path = /usr/lib/x86_64-linux-gnu/engines-3/libpkcs11.so
MODULE_PATH = /usr/lib/x86_64-linux-gnu/engines-3/pkcs11.so
init = 1

**Comment by JesusPoderoso** [comment]
Hello, @Jpgig 

Thank you for the report and for the fix! We will take a look at that thoroughly.
The documentation means to be installed on Ubuntu 20.04, so is that error happening while you tried to install Fast DDS in Ubuntu 22.04?

**Comment by Jpgig** [comment]
Yes, I am using Fast DDS in Ubuntu 22.04 LTS.
